lib/Model/index.ts

```diff
@@ -866,13 +866,17 @@ export class Model<T extends DocumentCarrier> {
 	}
 }
 
+export interface Model<T> {
+	new(doc?: Partial<T>): T;
+}
+
 Model.defaults = originalDefaults;
 
 
-Model.prototype.scan = function (this: Model<DocumentCarrier>, object?: ConditionInitalizer): Scan {
+Model.prototype.scan = function ( object?: ConditionInitalizer): Scan<DocumentCarrier> {
```

Comment:
Again, another thing that I wish the linter would enforce as an error haha.

```suggestion
Model.prototype.scan = function (object?: ConditionInitalizer): Scan<DocumentCarrier> {
```